web_tree_dynamic_colored_field/static/src/js/web_tree_dynamic_colored_field.js

```diff
@@ -17,15 +17,12 @@ odoo.define('web_tree_dynamic_colored_field', function (require) {
                 if (colorAttr.length > 0) {
                     var colorField = colorAttr[0].split(':')[1].trim();
                     // validate the presence of that field in tree view
-                    var fieldNames = _(this.columns).map(
-                        (value) => { return value.attrs.name; }
-                    );
-                    if (fieldNames.indexOf(colorField) === -1) {
+                    if (this.state.data.length && colorField in this.state.data[0].data) {
```

Comment:
if think `if (colorField in this.state.fields)` is enough, no?

Reply:
If others can confirm this i will change it.

Reply:
Ok... ```this.state.fields``` contains all fields of the model, ```this.state.data[0].data``` contains all the fields defined in the view for the first record.

Reply:
What is the final conclusion in this?

Reply:
@simahawk can you update your review?